neptune/client.py

```diff
@@ -269,6 +272,80 @@ def create_experiment(self,
             else:
                 raise
 
+    @with_api_exceptions_handler
+    def get_notebook(self, project, notebook_id):
+        try:
+            api_notebook_list = self.leaderboard_swagger_client.api.listNotebooks(
+                projectIdentifier=project.internal_id,
+                id=[notebook_id]
+            ).response().result
+
+            if not api_notebook_list.entries:
+                raise NotebookNotFound(notebook_id=notebook_id, project=project.full_id)
+
+            api_notebook = api_notebook_list.entries[0]
+
+            return Notebook(
+                client=self,
+                project=project,
+                _id=api_notebook.id,
+                owner=api_notebook.owner
+            )
+        except HTTPNotFound:
+            raise NotebookNotFound(notebook_id=notebook_id, project=project.full_id)
+
+    def get_last_checkpoint(self, project, notebook_id):
+        try:
+            api_checkpoint_list = self.leaderboard_swagger_client.api.listCheckpoints(
+                notebookId=notebook_id,
+                offset=0,
+                limit=1
+            ).response().result
+
+            if not api_checkpoint_list.entries:
+                raise NotebookNotFound(notebook_id=notebook_id, project=project.full_id)
+
+            return api_checkpoint_list.entries[0]
+        except HTTPNotFound:
+            raise NotebookNotFound(notebook_id=notebook_id, project=project.full_id)
+
+    @with_api_exceptions_handler
+    def create_notebook(self, project):
+        try:
+            api_notebook = self.leaderboard_swagger_client.api.createNotebook(
+                projectIdentifier=project.internal_id
+            ).response().result
+
+            return Notebook(
+                client=self,
+                project=project,
+                _id=api_notebook.id,
+                owner=api_notebook.owner
+            )
+        except HTTPNotFound:
+            raise ProjectNotFound(project_identifier=project.full_id)
+
+    @with_api_exceptions_handler
+    def create_checkpoint(self, notebook_id, jupyter_path, _file):
+        response = self._upload_raw_data(
+            api_method=self.leaderboard_swagger_client.api.createCheckpoint,
+            data=_file,
+            headers={"Content-Type": "application/octet-stream"},
+            path_params={
+                "notebookId": notebook_id
+            },
+            query_params={
+                "jupyterPath": jupyter_path
+            }
+        )
+
+        if response.status_code == NOT_FOUND:
+            raise NotebookNotFound(notebook_id=notebook_id)
+        elif response.status_code != OK:
+            raise RuntimeError("Http status code {}: {}".format(response.status_code, response.reason))
+
+        response.close()
```

Comment:
You should close resources in the finally block or use a context manager.

Reply:
This is fixed in next pull request.